### PR TITLE
Added development plan document

### PR DESCRIPTION
**Pull request by oruebel**

### CODE REVIEW

File: docs/general/source/development_plan.rst [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+
+The following sections provide an overview of various development targets for NWB:N and what they entail.
+
+.. _dev-nwb2beta:
+
+NWB:N 2.0 beta (11/10/2017)
+---------------------------
+
+This is a first public beta release of the new version NWB:N. The stability of all modules can't be ensured. Backward compatibility cannot be promised at this point yet. Further changes to the APIs as well as the format are still planed between this beta and the first full release of NWB 2.0 (see Section :ref:`dev-nwb2` for details). The intent of this beta release is to enable early adopters to start exploring the new format and software.
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: `The stability of all modules can't be ensured.` what do you mean by this? should be more precise.

Comment (mgrauer) [reply docs/general/source/development_plan.rst]: s/planed/planned/

File: docs/general/source/development_plan.rst [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        * Read NWB:N files (base round-trip read/write)
+        * Validate NWB:N files
+        * Create new format extensions
+        * Read/Write data using extensions
+        * Read legacy NWB 1.0x files from the Allen Institute for Brain Science
+
+    * **Main contributors for this release** :
+
+        * Andrew Tritt, Oliver Ruebel, David Camp, Kris Bouchard (LBNL) : Lead architects and development team
+        * Nicholas Chain (Allen Institute for Brain Science) and Andrew Tritt (LBNL) : Port to Python 2 and legacy read for Allen data
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/Chain/Cain/

File: docs/general/source/development_plan.rst [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        * Codecov.io
+
+    * **Deployment**:
+        * PIP (PyPI)
+        * Conda (CondaCloud)
+
+    * **Main contributors for this release**
+
+        * Andrew Tritt (LBNL) : Lead for design and implementation of the Python test suites
+        * Jean-Christophe Fillion-Robin, Doruk Ozturk, Chris Kotfila Michael Grauer, Will Schroeder (Kitware) : Lead development team for creation of continuous integration, testing, and deployment mechanisms
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/Kotfila Michael/Kotfila, Michael/

File: docs/general/source/development_plan.rst [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+* **Documentation** We have created dedicated online documentations for the various aspects of NWB:N. While the documents are quite extensive, completeness and consistency of the documents is not guaranteed for the beta release and examples, tutorials and release notes (i.e., changes), may be a bit out of date given pace of development.
+
+    * **Available online**:
+        * General overview of NWB:N and its various components: http://nwb-overview.readthedocs.io/en/latest/nwbintro.html
+        * PyNWB API for NWB:N: http://pynwb.readthedocs.io
+        * NWB:N data format specification: http://nwb-schema.readthedocs.io
+        * Specification Language: http://schema-language.readthedocs.io
+        * Data Storage: http://nwb-storage.readthedocs.io
+
+    * **Format Documentation Tools** To ensure consistency between the NWB:N format specification and documentation we have developed a set of tools generate Sphinx RST documents from the YAML specification sources. The tools are available in `docs/utils` as part of the nwb-schema repository for Python 2/3.
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/tools generate/tools that generate/

File: docs/general/source/development_plan.rst [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+* **PyNWB:** https://github.com/NeurodataWithoutBorders/pynwb/issues?q=is%3Aopen+is%3Aissue+milestone%3Asfn_release
+* **nwb-schema:** https://github.com/NeurodataWithoutBorders/nwb-schema/issues?q=is%3Aopen+is%3Aissue+milestone%3Asfn_release
+
+
+.. _dev-nwb2:
+
+NWB:N 2.0 (~June 2018)
+----------------------
+
+The primary purpose of this targe release is to stabilize and complete development of NWB:N 2.0beta. This release will also include a number of changes to the format and APIs that have not made it into the beta release. Request for new features for this release has been frozen. Additional new features may be included in future NWB:N 2.x releases.
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/targe/target/
s/Request/Requests/

File: docs/general/source/development_plan.rst [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+Planed new features (beyond NWB:N 2.0 beta)
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: Planned

File: docs/general/source/development_plan.rst [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **NWB:N Format Specification (and PyNWB API)**
+
+    * Support saving of NWB:N specifications in NWB:N HDF5 files as part of the ``/general/specifications``. Support use of these cached specification for data read.
+    * Reorganize ``/general`` to ease additiona and management of custom, lab-specific metadata via extensions
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/additiona/additional/

but additional what?

File: docs/general/source/development_plan.rst [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+
+* **NWB:N Specification Language (and PyNWB API)**
+
+*   * No major new features are planned for the specification language itself that have not been added in NWB:N 2.0 beta.
+
+* **PyNWB**
+
+    * Expand legacy read support to more sets of Allen data
+    * Expand and refine existing features for read/write/specfication etc.
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: specification

File: docs/general/source/development_plan.rst [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+* **Testing, Continuous Integration, Deployment**
+
+    * Add integration test cases to ensure broad coverage of intergration tests
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/intergration/integration/

File: docs/general/source/development_plan.rst [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **PyNWB:** https://github.com/NeurodataWithoutBorders/pynwb/issues?q=is%3Aopen+is%3Aissue+milestone%3AFuture
+* **nwb-schema:** https://github.com/NeurodataWithoutBorders/nwb-schema/issues?q=is%3Aopen+is%3Aissue+milestone%3AFuture
+
+
+Internal GitHub Project Board
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following project board organizes issues related to NWB 2.x from the various repos: https://github.com/orgs/NeurodataWithoutBorders/projects/2  Note, as this is an organization-level project board, this is unfortunately only available to members of the NeurodataWithoutBorders GitHub organization. Note, until work begins on the NWB:N 2.x we may occasionally add nodes and tickets to this board to help with the planning and organization for NWB 2.x later on, however, this board is currently not activily maintained and may not reflect actual future plans for NWB 2.x.
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/nodes/notes/ , at least I think so

File: docs/general/source/development_plan.rst [hunk 11]
```diff
@@ -0,0 +1,225 @@
+.. _dev-plan:
+
+Development Plan
+================
+
+.. todo::
+
+    This is a first draft. Check completeness and consistency of all items. Check and approbriateness of  timelines for the release.
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: s/approbriateness/appropriateness/

File: docs/general/source/development_plan.rst [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+
+    * **Core features**:
+
+        * Write NWB:N HDF5 files
+        * Read NWB:N HDF5 files
+        * Read/Write data for custom extensions
+
+    * **Main contributors for this release:**
+        * Nathan Clack and Lawrence Niu (Vidriotech) : Software design and development lead
+        * Karel Svoboda : Project Lead
```
Comment (mgrauer) [inline docs/general/source/development_plan.rst]: affiliation?

Comment (mgrauer) [reply docs/general/source/development_plan.rst]: maybe `Janelia Farms HHMI`

**Review by mgrauer — CHANGES_REQUESTED** [review]
Generally looks good, the content is right, just some typos and a few small changes requested.

File: docs/general/source/development_plan.rst (hunk 8 above)
Comment (oruebel) [reply docs/general/source/development_plan.rst]: @mgrauer What change does this comment refer to?

**Comment by oruebel** [comment]
@mgrauer thanks for the review and comments! I've addressed the changes you suggested in https://github.com/NeurodataWithoutBorders/nwb-schema/pull/80/commits/04e06d01d43e484943a5b67686a75f96e7f8de75  There was one comment where I couldn't tell what it was referring to (see comment above). If you let me know what that missing edit is, then I'll add it in.

File: docs/general/source/development_plan.rst (hunk 8 above)
Comment (mgrauer) [reply docs/general/source/development_plan.rst]: sorry @oruebel 

s/specfication etc./specification etc./

Comment (oruebel) [reply docs/general/source/development_plan.rst]: Thanks @mgrauer. Fixed now.

**Review by mgrauer — APPROVED** [review]

**Pull request merged**